### PR TITLE
[OpenMP] Add support for Solaris/x86_64

### PR DESCRIPTION
**Pull request by brad0**
Tested on `amd64-pc-solaris2.11`.

### CODE REVIEW

**Comment by shiltian** [comment]
I'm not familiar with Solaris but does it need dedicated implementation of the function invocation written in ASM?

**Comment by brad0** [comment]
> I'm not familiar with Solaris but does it need dedicated implementation of the function invocation written in ASM?

Can you point out what you're referring to? Looking at the patches for adding support for OpenBSD and other OS's I don't see any ASM additions.

**Comment by shiltian** [comment]
> > I'm not familiar with Solaris but does it need dedicated implementation of the function invocation written in ASM?
> 
> Can you point out what you're referring to? Looking at the patches for adding support for OpenBSD and other OS's I don't see any ASM additions.

I was thinking about `openmp/runtime/src/z_Linux_asm.S` but it looks like we don't need extra stuff there.

**Comment by brad0** [comment]
I'd like to get this in.

**Review by shiltian — APPROVED** [review]
LG

File: openmp/runtime/src/z_Linux_util.cpp [hunk 1]
```diff
@@ -70,6 +72,15 @@ struct kmp_sys_timer {
   struct timespec start;
 };
 
+#ifdef KMP_OS_SOLARIS
+// Convert timeval to timespec.
+#define TIMEVAL_TO_TIMESPEC(tv, ts)                                            \
```
Comment (shiltian) [inline openmp/runtime/src/z_Linux_util.cpp:77]: It looks like this macro is not guarded correct somewhere else such that the compiler complains on a non-Solaris system.
```
[19/128] Building CXX object runtime/src/CMakeFiles/omp.dir/z_Linux_util.cpp.o
/home/ac.shilei.tian/Documents/vscode/llvm-project/openmp/runtime/src/z_Linux_util.cpp:77:9: warning: 'TIMEVAL_TO_TIMESPEC' macro redefined [-Wmacro-redefined]
   77 | #define TIMEVAL_TO_TIMESPEC(tv, ts)                                            \
      |         ^
/usr/include/sys/time.h:38:10: note: previous definition is here
   38 | # define TIMEVAL_TO_TIMESPEC(tv, ts) {                                   \
      |          ^
```

**Pull request merged**